### ISSUE TITLE
chromeos is stoned while Linux is fine.

### ISSUE THREAD

**Issue by midnightradio** [dnschneid/crouton#1513]
It may not be an issue for crouton, I know, but there's no other sites where I can expect some nice responses for this. ChromeOS guys won't pay any attention since the issue never happens without running Linux at the same time, and so as Linux guys for the same reason. Would be thankful if you read it to the end and any kind of suggestions will be appreciated.

The problem is ChromeOS just being stalled in some circumstances while I'm running Linux over crouton. It started to happen about 10 days ago, and as I remember, it started just before I updated current stable version (41.0.2272.89) and therefore it may not related to it (Or. maybe related to the previous before the current which introduced a little earlier than the current). Actually, this issue made me to update crouton because at that time I haven't updated crouton since Dec last year and simply I suspected crouton as a cause. So I updated, faced with another issue rebooting (#1490) , and came back to life by crouton guys' help, however, I'm still having the same trouble with my chromeos getting stoned.

Here the status "stalled" means it allows nothing other than keyboard and mouse functioning. When I meet the problem with my browser running, I can add a tab (ctrl+t) or key in a URL to existing tabs but cannot expect any responses after that as if I lost a network connection. I can still check the antenna in normal shape, and as you know how chromebook reacts when it is disconnected it doesn't even print out on the browser the dragon saying you are disconnected. And everything on Linux are fine so I can continue with my jobs on Linux VT screen just leaving my chromeos fool. Running "Settings" won't display nothing but its' widget, and running an app such as "Files" doesn't even display a window.

After experiencing the issue for about a week, now I learned under what circumstances does the issue happens and I can reproduce it. It seems very closely related to IDLE state of a Linux while my screen is on Linux side. If Linux goes to idle and then I can meet my stoned chromeos by waking Linux up and going back to chrome side. To be more specific, with a Linux screen displayed, when I close the lid or just leave my laptop (chromebook pixel) alone allowing it to sleep, I can later wake it up and do my jobs on Linux but I have to reboot my machine to do something on chrome side. I have no screen saver on Linux, and it never happens for closing a lid or leaving it to sleep on Chrome side. Therefore, I always have to go back to chrome side before closing the lid or leaving it goes to idle.

I only use xfce though I have installed gnome and xiwi. 
```
(trusty)lyle@localhost:~$ croutonversion
crouton: version 1-20150304145447~master:550afdb3
release: trusty
architecture: amd64
xmethod: xorg
targets: gnome,xorg,keyboard,audio,cli-extra,xiwi,extension,touch,xfce
host: version 6680.64.0 (Official Build) stable-channel link
(trusty)lyle@localhost:~$
```

![screenshot 2015-03-19 at 7 04 51 pm](https://cloud.githubusercontent.com/assets/149314/6729432/397f1efc-ce6d-11e4-8e92-d8695815ab04.png)

**Comment by DennisLfromGA** [comment]
That's interesting, luckily that hasn't happenned to me as yet.

I'm wondering if a 'reboot' of the chrome browser would kickstart it. Kind of like when you change a chrome flag in `chrome://flags` and then click on the 'Restart Now' tab at the bottom of the screen.
You can change any flag then undo the chamge to get the 'Restart Now' button. 
Be careful though as this may kill your chroot also, I'm not sure - it may be worth a try.

I know there's a way to do this via the command line but I can't remember if it's just `sudo killall chrome` or what it is exactly...

**Comment by xeoron** [comment]
I am wondering if you considered disabling your Chrome Extensions? I have
had ChromeOS freeze after the system updates and 1 or more the extension
was the cause.

-Jason

**Comment by midnightradio** [comment]
@DennisLfromGA Thank you very much for responding and also for some tries on finding a way to kickstart. I prefer Xorg to Xiwi due to it's better performance and definitely it is worth a try. I'll try when I get safe with my sessions on Linux. :+1: 

@xeoron You are right, extensions are suspicious on all kinds of weird happenings on Chromebook. Actually, I already tried with minimizing the number of enabled extensions but maybe I can try more by disabling all extension at a time and checking if I can reproduce the issue. Good point. Thank you.

**Comment by midnightradio** [comment]
@DennisLfromGA here's a good news and a bad news. 
I confirmed "Restart Now" on chrome://flags does not hurt the other side and I could preserve Linux/Xorg screen intact. However, when chromeos is in vegitated state, it doesn't allow me to access to the page, chrome://flags and cannot try to restart the zombie. 

Anyway, that was a clever idea and might be useful to another situation in future. Thanks!

**Comment by midnightradio** [comment]
@xeoron No, none of extensions causing the trouble as chromsos becomes zombie still after turning off all extensions and reproducing it. At least now I can continue to love those extensions without putting any guilt on them. Thanks. :-)

**Comment by divx118** [comment]
@midnightradio You could try and see if anything is eating resources by running `top` in a shell. Probably not, because your chroot seems to work fine, but you never know. Also just to be sure I would look at `dmesg` if you see anything out of the order regarding sleep and coming out of standby.

**Comment by midnightradio** [comment]
@divx118 A shell left running on chromeos before it gets stalled also frozen and does not give any valuable information. (I could capture this and surprised that screen capture is functioning) Also there's nothing interesting consuming resources can be found on ```top``` in a shell on linux side.

Lastly, ```dmesg``` shows more than 3K lines and ... you know ... 

Anyway, thanks for showing your interest though.

![screenshot 2015-03-21 at 1 21 12 am](https://cloud.githubusercontent.com/assets/149314/6757268/c6daa390-cf6a-11e4-829e-47b44a73df80.png)
![screenshot - 03212015 - 01 22 15 am](https://cloud.githubusercontent.com/assets/149314/6757278/d6b5eff4-cf6a-11e4-82ec-ebfada0de15f.png)

**Comment by zmanian** [comment]
I've been seeing stalled state after wake if the OS suspends while the
crouton virtual terminal is on screen. If the Chrome Os virtual terminal is
onscreen, sleep & wake don't produce the stalled effect.

This appears to be fairly repeatable on my Gen 1 Pixel.

**Comment by midnightradio** [comment]
@zmanian Welcome to the club! Glad to have a company! Let's find a solution together.

**Comment by kenschnall** [comment]
@midnightradio Try the command 'sudo stop powerd' inside your crouton os, then see what happens when you try to trigger the behavior.

For me, closing the lid in my crouton Unity operating system would cause Chrome OS to stone out like you're saying.  Using this command stopped that stoner behavior.

**Comment by midnightradio** [comment]
@kenschnall That works, but do you know how it make the difference? I mean, under the situation that I killed powerd process and closed the lid in linux system, the blue light on the back of the lid never turns off and presumably it's consuming power not just for the blue light but for whole system. Not only that, with opening the lid again and going back to chromeos, I can see without powerd chromeos doesn't automatically log off. (of course it does still when I close the lid in chromeos)

It's really good to know having another company getting a tip. I wonder if I can fix it by replacing the powerd or re-configuring it but not by just killing it.

**Comment by midnightradio** [comment]
I don't need to close the lid for reproducing the problem anymore but just need to run following command after running startcli, startxfce4 or startgnome.
```
$ /usr/local/bin/croutonpowerd --suspend
```
or
```
$ host-dbus dbus-send --system --dest=org.chromium.PowerManager \
--type=method_call /org/chromium/PowerManager \
org.chromium.PowerManager.RequestSuspend
``` 
However, it doesn't cause stalled ChomeOS when I run this command on Crosh or on a shell after running enter-chroot as follows.
```
chronos@localhost / $ dbus-send --system --dest=org.chromium.PowerManager \
--type=method_call /org/chromium/PowerManager \
org.chromium.PowerManager.RequestSuspend
chronos@localhost / $ sudo enter-chroot
Password: 
Enter encryption passphrase for trusty: 
Entering /mnt/stateful_partition/crouton/chroots/trusty...
(trusty)lyle@localhost:~$ host-dbus dbus-send --system --dest=org.chromium.PowerManager \
--type=method_call /org/chromium/PowerManager \
org.chromium.PowerManager.RequestSuspend
```
I suspect that passing RequestSuspend to PowerManager from VT2 causing the suspending problem on VT1. I'm gonna report this issue to chromium.

**Comment by ArchimedesPi** [comment]
I just started having this problem a couple days ago. I'm on the beta release channel, but I can powerwash back to stable and see if that fixes stuff for now.

**Comment by ArchimedesPi** [comment]
Yep, downgrading to stable fixes it for me.

**Comment by midnightradio** [comment]
@ArchimedesPi Glad you could fix your problem. Can you tell us which stable version are you in? 41.0.2272.89, right? Please share which chromebook you have or some more information with running command croutonversion. Thanks.

**Comment by ArchimedesPi** [comment]
I'm on a Toshiba Chromebook 2:
Version 41.0.2272.89 (64-bit)
Platform 6680.64.0 (Official Build) stable-channel swanky
Firmware Google_Swanky.5216.238.5

**Comment by midnightradio** [comment]
How about crouton version? 
Ive always in stable version with my pixel and nowhere to go back to fix the problem as you did. Seems it only happens to pixel.

**Comment by ArchimedesPi** [comment]
I'm using Croagh right now, which is based on Crouton from 2013ish, but I used to have the same problems with both Croagh and Crouton (head).

**Comment by midnightradio** [comment]
Maybe the related issue in chromium os issue tracker needs more stars to be addressed.
https://code.google.com/p/chromium/issues/detail?id=469406

**Comment by MAkerboom** [comment]
I noticed chromeos tot be stoned also, i thought unchecking the chromeos option to lock on suspend in chromeos fixed it. Can you try?

**Comment by midnightradio** [comment]
@MAkerboom would you please teach me where I can find such option, "lock on suspend"?

**Comment by MAkerboom** [comment]
https://support.google.com/chromebook/answer/2587994

- If you haven't already, sign in to your Chromebook.
- Click the status area in the lower-right corner, where your account picture appears.
- Select Settings and find the "Users" section.
- Use the "Require password to wake from sleep" checkbox to adjust this setting.

**Comment by midnightradio** [comment]
@MAkerboom It worked! Thanks for a nice workaround. It makes me feel more comfortable with not being too much conscious of not closing the lid while working on VT2. Also I think it can be a hint for locating the root cause. :+1:

**Comment by drinkcat** [comment]
Okay, for those who have this problem, can you update their chroot, and provide the output of `croutonversion`?

Trying to figure out if this is freon-related... Thanks.

**Comment by MAkerboom** [comment]
Mine has Freon: yes
(not with my chromebook right now, will update the post with full output)

**Comment by midnightradio** [comment]
Mine is not Freon. Though I don't know what that is, I know how to check on Freon. (CTRL+ALT+->; "freon" on top) So is not related to that whatever that is as I'm still experiencing the problem.

**Comment by Desertions** [comment]
Issue here aswell, also happens upon booting up and trying tot log-in, 90% of the time I can't even use my keyboard after a second of logging in. so I have to keep resetting my chromebook 20~ times until it works.

**Comment by tedm** [comment]
help me out here. What does "stoned" mean?

**Comment by midnightradio** [comment]
Hi, @tedm . Now I can specify the status I wrote "stoned" as the condition does not wake up after being suspended and trying to log on. So far, the best workaround is un-checking the option "require password to wake from sleep" so that the system not to ask any password when it wakes up.

**Comment by tedm** [comment]
@midnightradio Thanks for the clarification. I think it may be fixed now. I entered a chroot and ran top. Checked the setting to require a password in Chrome, closed the lid, opened it, entered password, and top in the chroot resumed. This is on the ARM, stable channel, crouton from this morning, and kali chroot from yesterday.

**Comment by Watney** [comment]
@midnightradio [off topic] thanks for the explanation. Us [old people](http://www.urbandictionary.com/define.php?term=old+people&defid=1935378) need a dictionary to keep up with the vocab on this site!

**Comment by midnightradio** [comment]
@tedm The day before yesterday, I checked with a new CROS update and confirmed having the problem still. It happens to not all but only a few chromebooks and also it only happens when you are on VT2 but not in a condition when you just enter chroot with crosh running on VT1. Maybe yours never had the problem. :+1: I appreciate your help with checking on your chromebook.

@Watney [off topic] Sorry for my ambiguous word selection. Good to have a chance to make up for better understanding , and it may help with fixing the problem.

**Comment by jcward** [comment]
Grr, same here on an Acer C720, Google Chrome 41.0.2272.102 (Official Build). Turning off security / lockscreen doesn't seem like a nice workaround.  :(

**Comment by ParkerCraftLite** [comment]
Same problem here, Toshiba Chromebook 2.

**Comment by midnightradio** [comment]
Welcome to the club! Each of you who experiencing the problem would better to add a push (star) to chromium (https://code.google.com/p/chromium/issues/detail?id=469406).

@ParkerCraftLite , In my cast, there was no response after 3 and cannot do anything on VT1 other than power off. Just turning off "lockscreen" works fine preventing me from the problem.

**Comment by dnschneid** [comment]
@midnightradio I suspect Freon will solve your issues, and pragmatically, the devs are probably focusing on stabilizing Freon and getting it pushed to you than fixing long-tail issues with VTs (which Freon completely replaces).  Have you tried beta or dev channels to see if your platform has been freonified?

**Comment by midnightradio** [comment]
@dnschneid Thanks for your interest. Actually I've never tried beta or dev channels but I hope I could...

**Comment by dnschneid** [comment]
It's easy to switch to them, but it's hard(ish) to switch back to stable (you have to powerwash, so make a backup of your chroot first).

**Comment by tonyxue** [comment]
@dnschneid But is the new VT thing in dev channel working well right now? Seems it is still broken in #706 for what I have experienced. Are these similar issues?

**Comment by dnschneid** [comment]
I think the separate partition script needs to be updated, but mainline crouton works fine.

**Comment by midnightradio** [comment]
I guess this issue has been resolved.

I updated CrOS to latest,
```
Version 42.0.2311.87 (64-bit)
Platform 6812.75.2 (Official Build) stable-channel link
Firmware Google_Link.2695.1.156
```
and faced with a problem failing on running startxfce4. And I was reluctant to update crouton as well since there was no other choice. I couldn't do anything with crouton since I had this issue and found workaround solution not to risk my work mainly have been done with my chrome linux. 

Eventually, I'm on recent versions of both crouton and CrOS, and checked whether still the problem happens with suspending. No, It's gone now. No need to disable password lock in CrOS. 
```
crouton: version 1-20150414182244~master:da478a77
release: trusty
architecture: amd64
xmethod: xorg
targets: gnome,xorg,keyboard,audio,cli-extra,xiwi,extension,touch,xfce
host: version 6812.75.2 (Official Build) stable-channel link 
kernel: Linux localhost 3.8.11 #1 SMP Tue Apr 14 21:10:03 PDT 2015 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
```

Those who have been suffered from the same issue may try to update the system and please let us know.

**Comment by dnschneid** [comment]
Welcome to Freon :)

**Comment by jcward** [comment]
FYI, upgrading Chrome and crouton fixed it for me as well. Nice to be able to keep my mobile laptop locked. Here's what I did to upgrade:

At chrome://chrome click "Check for and Apply Updates", reboot. At this point, my crouton installation was unusable, segfault when starting `/usr/bin/X`

Then I followed the instructions [here](http://www.howtogeek.com/210047/how-to-manage-the-crouton-linux-system-on-your-chromebook/) to upgrade crouton, and all is well again.

Notes in case you've forgotten: `ctrl-alt-t` opens the crosh shell, then `shell` opens the unix shell, and `sudo enter-chroot` is how you enter the chroot. FYI, the upgrade took about 20 minutes for Ubuntu trusty with Unity (lots of package downloads/installs.)
